src/main/java/com/scalar/database/transaction/consensuscommit/RecoveryHandler.java

```diff
@@ -29,7 +33,19 @@ public RecoveryHandler(DistributedStorage storage, Coordinator coordinator) {
     this.coordinator = checkNotNull(coordinator);
   }
 
-  // lazy recovery in read phase
+  /**
+   * This method rolls back or rolls forward a record according to the state of the corresponding
+   * transaction.
+   *
+   * <p>Given a specified {@link Selection} and {@link TransactionResult} will first check the
+   * {@link TransactionState} of the result. If the state of the transaction is committed then the
+   * selection will be rolled forward and if not then rolled back.
```

Comment:
I don't understand "`selection` will be rolled forward". It is an operation.

Reply:
Thank you! Updated it somewhat.